src/librbd/ImageState.cc

```diff
@@ -152,6 +311,45 @@ void ImageState<I>::snap_set(const std::string &snap_name, Context *on_finish) {
 }
 
 template <typename I>
+int ImageState<I>::register_update_watcher(UpdateWatchCtx *watcher,
+					 uint64_t *handle) {
+  CephContext *cct = m_image_ctx->cct;
+  ldout(cct, 20) << __func__ << dendl;
+
+  Mutex::Locker locker(m_lock);
+  if (m_update_watchers == nullptr) {
```

Comment:
Is it possible for the user to have access to the image before it's fully opened?  Perhaps an assert if not.